### DISCUSSION TITLE
Select belongsTo relationships cause multiple model events being fired

### DISCUSSION THREAD

**Discussion by boris-glumpler** [filamentphp/filament#7618] [Help] [answered]
Hello,

noticed this weird issue recently where model events would fire multiple times when saving a resource form. Sometimes twice, other forms up to 4 times. So, dug into it a bit and it turns out that `Select` components not only associate the `belongsTo` relationship to the record but then also save the record. This means that model events are being run once for each relationship you might have on a form.

Wouldn't it make more sense to just associate the relationship and leave the saving up to the form? Or at least save the record quietly if the save operation is somehow necessary. Either would solve the issue of multiple model events being fired.

Would you consider a PR for this? Cheers!

**Comment by danharrin** [comment, marked as answer]
Thanks for bringing this to my attention, it will be fixed in the next release https://github.com/filamentphp/filament/commit/6dcb86ce81cafb82f2c82fd06cd4f802143e64fc

**Comment by boris-glumpler** [comment]
Cheers, that was quick! I'm still on v2 for at least another couple months, I guess. Will it be back-ported?

**Reply by danharrin** [reply]
In v2, you can patch it using `Select::make()->saveRelationshipsUsing(null)`. Then you can remove that in your v3 upgrade, so make a note of that.
